src/functions/formatSort.js

```diff
@@ -5,22 +5,38 @@ var find = require('./find');
 /**
  * Transform sort format from user friendly notation to lodash format
  * @param {string[]} sortBy array of predicate of the form "attribute:order"
+ * @param {string[]} [defaults] array of predicate of the form "attribute:order"
  * @return {array.<string[]>} array containing 2 elements : attributes, orders
  */
 module.exports = function formatSort(sortBy, defaults) {
+  var defaultInstructions = (defaults || []).map(function(sort) {
+    return sort.split(':');
+  });
+
   return sortBy.reduce(
-    function preparePredicate(out, sortInstruction) {
-      var sortInstructions = sortInstruction.split(':');
-      if (defaults && sortInstructions.length === 1) {
-        var similarDefault = find(defaults, function(predicate) {
-          return predicate[0] === sortInstruction[0];
-        });
-        if (similarDefault) {
-          sortInstructions = similarDefault.split(':');
-        }
+    function preparePredicate(out, sort) {
+      var sortInstruction = sort.split(':');
+
+      if (sortInstruction.length > 1) {
+        out[0].push(sortInstruction[0]);
+        out[1].push(sortInstruction[1]);
+        return out;
+      }
+
+      var matchingDefault = find(defaultInstructions, function(
```

Comment:
We can move this assignation above the `if`. We can then group the condition to only use two `return`. We might compute the matching default for nothing but I don't think it matters that much. The overall function would be easier to understand.

Reply:
like this?

```js
      var sortInstruction = sort.split(':');
      
      var matchingDefault = find(defaultInstructions, function(
        defaultInstruction
      ) {
        return defaultInstruction[0] === sortInstruction[0];
      });

      if (sortInstruction.length > 1 || !matchingDefault) {
        out[0].push(sortInstruction[0]);
        out[1].push(sortInstruction[1]);
        return out;
      }

      out[0].push(matchingDefault[0]);
      out[1].push(matchingDefault[1]);
      return out;
```